dp/generate_climos.py

```diff
@@ -61,34 +60,48 @@ def input_check(filepath, climo):
         return None, []
 
     periods = input_file.climo_periods.keys() & climo
-    logger.info("climo_periods: {}".format(periods))
+    outputer(f"climo_periods: {periods}")
     if len(periods) == 0:
-        logger.critical(f"{input_file.filepath()} contains no standard climatological periods")
+        logger.critical(
+            f"{input_file.filepath()} contains no standard climatological periods"
+        )
```

Comment:
I am not so sure about what you meant by `our intention of generalizing the output method`. Did you mean it is against the format of `dry_run` logging such as `attribute_key: attribute_value`? If so, I think it is OK the way it is since `input_check` is not only called in `dry_run_handler` but also the general `generate_climos`, and the messages are to tell what problems caused the error, which is different from the purpose of other logging messages in `dry_run_handler`.

Reply:
Reading back my comment I don't think I made it very clear what my concern is.

We want to use `outputer` to generalize how we are logging/writing the messages. The method shouldn't care what the `outputer` actually is, only that it takes a single argument that is a string and we expect it to be stored/written somewhere. If we are using `logging.info` we can't log messages that are higher levels than that (i.e. `critical`). If we want to use higher levels we need to sort out a generic way of going about it.

Does that clear things up a bit?

Reply:
>If we are using logging.info we can't log messages that are higher levels than that (i.e. critical). If we want to use higher levels we need to sort out a generic way of going about it.

@nikola-rados I am still not sure if I am following. Do you want the messages logged with `CRITICAL` level shown with the `dry-run` info messages, which are logged with `INFO` level? If so, I am seeing both messages when I ran the following commands.
```
(venv) slim@devel2:~/Desktop/climate-explorer-data-prep$ generate_climos -o output ./tests/data/tiny_daily_pr.c -p mean -n -f
-->> Could not load xarray!! <<--
2020-07-21 10:56:26 CRITICAL: OSError: No such file or directory
Traceback (most recent call last):
  File "/home/slim/Desktop/climate-explorer-data-prep/venv/bin/generate_climos", line 7, in <module>
    exec(compile(f.read(), __file__, 'exec'))
  File "/home/slim/Desktop/climate-explorer-data-prep/scripts/generate_climos", line 116, in <module>
    main(args)
  File "/home/slim/Desktop/climate-explorer-data-prep/scripts/generate_climos", line 21, in main
    dry_run_handler(filepath, args.climo, outpath)
  File "/home/slim/Desktop/climate-explorer-data-prep/dp/generate_climos.py", line 97, in dry_run_handler
    outputer(f"dependent_varnames: {input_file.dependent_varnames()}")
AttributeError: 'NoneType' object has no attribute 'dependent_varnames'
```
```
(venv) slim@devel2:~/Desktop/climate-explorer-data-prep$ generate_climos -o output ./tests/data/tiny_downscaled_pr.nc -p mean --climo 2080 -n
-->> Could not load xarray!! <<--
2020-07-21 11:08:31 INFO: DRY_RUN
2020-07-21 11:08:31 INFO: File: ./tests/data/tiny_downscaled_pr.nc
2020-07-21 11:08:31 INFO: climo_periods: set()
2020-07-21 11:08:31 CRITICAL: ./tests/data/tiny_downscaled_pr.nc contains no standard climatological periods ['2080']
2020-07-21 11:08:31 INFO: project: AttributeError: 'NoneType' object has no attribute 'metadata'
2020-07-21 11:08:31 INFO: institution: AttributeError: 'NoneType' object has no attribute 'metadata'
2020-07-21 11:08:31 INFO: model: AttributeError: 'NoneType' object has no attribute 'metadata'
2020-07-21 11:08:31 INFO: emissions: AttributeError: 'NoneType' object has no attribute 'metadata'
2020-07-21 11:08:31 INFO: run: AttributeError: 'NoneType' object has no attribute 'metadata'
Traceback (most recent call last):
  File "/home/slim/Desktop/climate-explorer-data-prep/venv/bin/generate_climos", line 7, in <module>
    exec(compile(f.read(), __file__, 'exec'))
  File "/home/slim/Desktop/climate-explorer-data-prep/scripts/generate_climos", line 116, in <module>
    main(args)
  File "/home/slim/Desktop/climate-explorer-data-prep/scripts/generate_climos", line 21, in main
    dry_run_handler(filepath, args.climo, outpath)
  File "/home/slim/Desktop/climate-explorer-data-prep/dp/generate_climos.py", line 97, in dry_run_handler
    outputer(f"dependent_varnames: {input_file.dependent_varnames()}")
AttributeError: 'NoneType' object has no attribute 'dependent_varnames'
```